### ISSUE TITLE
Launcher should only re-run unsuccessful runs

### ISSUE THREAD

**Issue by prabhuramachandran** [ioam/lancet#10]
Launcher currently re-launches everything when called again.  Instead Launcher should only run previous runs if the earlier invocation failed (with a non-zero exit status) or explicitly asked to force-rerun.  This would also make it easy to change the parameters and add additional parameters ranges to explore without having to worry about re-running everything.

**Comment by melver** [comment]
Similar to feature request #9, this sounds useful, but is not realizable for other instances of Launcher, e.g. QLauncher.

The exit-code is also not a one-size-fits-all solution, as there are cases where a range of exit-codes, not just zero, are valid (e.g. some of my experiments have this property). A more generic solution would be to have an application-specific way to detect failure of a previous invocation, and then either apply a filter or the proposed subtraction #5. Finally, the way your process the results would then be able to merge multiple invocations' results.

**Comment by jlstevens** [comment]
I agree that this feature request is potentially quite thorny and problematic. In particular, exit codes won't work for ``QLauncher`` as melver points out so you would need some other way of finding out whether a job failed or not.

I also worry about the idea of merging multiple invocations due to version control reasons but isn't this already possible by disabling the timestamp? 

@melver Isn't that what you do yourself? Don't you ever re-launch some jobs with the same parameters as launched previously e.g to fill in jobs that might have mysteriously failed and have the results accumulate in the same directory as a previous launch?

**Comment by melver** [comment]
On Sun, 10 May 2015, J.L Stevens wrote:

> @melver Isn't that what you do yourself? Don't you ever re-launch some jobs with the same parameters as launched previously e.g to fill in jobs that might have mysteriously failed and have the results accumulate in the same directory as a previous launch?

I don't, never disabled timestamps.

**Comment by jlstevens** [comment]
> I don't, never disabled timestamps.

Hmm. Maybe not any more but I am sure you requested this at some point (i.e the distant past!). Should check that this still works (maybe even write some tests for it).
